packages/app-mobile/services/e2ee/crypto.ts

```diff
@@ -0,0 +1,151 @@
+import { _ } from '@joplin/lib/locale';
+import { Crypto, CryptoBuffer, Digest, CipherAlgorithm, EncryptionResult } from '@joplin/lib/services/e2ee/types';
+import QuickCrypto from 'react-native-quick-crypto';
+import { HashAlgorithm } from 'react-native-quick-crypto/lib/typescript/keys';
+import type { CipherGCMOptions, CipherGCM, DecipherGCM } from 'crypto';
+
+type DigestNameMap = Record<Digest, HashAlgorithm>;
+
+const crypto: Crypto = {
+
+	getCiphers: () => {
+		return QuickCrypto.getCiphers();
+	},
+
+	getHashes: () => {
+		return QuickCrypto.getHashes();
+	},
+
+	randomBytes: async (size: number) => {
+		return new Promise((resolve, reject) => {
+			QuickCrypto.randomBytes(size, (error, result) => {
+				if (error) {
+					reject(error);
+				} else {
+					resolve(result);
+				}
+			});
+		});
+	},
+
+	pbkdf2Raw: async (password: string, salt: CryptoBuffer, iterations: number, keylen: number, digest: Digest) => {
+		const digestNameMap: DigestNameMap = {
+			sha1: 'SHA-1',
+			sha224: 'SHA-224',
+			sha256: 'SHA-256',
+			sha384: 'SHA-384',
+			sha512: 'SHA-512',
+			ripemd160: 'RIPEMD-160',
+		};
+		const rnqcDigestName = digestNameMap[digest];
+
+		return new Promise((resolve, reject) => {
+			QuickCrypto.pbkdf2(password, salt, iterations, keylen, rnqcDigestName, (error, result) => {
+				if (error) {
+					reject(error);
+				} else {
+					resolve(result);
+				}
+			});
+		});
+	},
+
+	encryptRaw: (data: CryptoBuffer, algorithm: CipherAlgorithm, key: CryptoBuffer, iv: CryptoBuffer | null, authTagLength: number, associatedData: CryptoBuffer | null) => {
```

Comment:
<details><summary>🔵 Optional/structure: Don't export <code>encryptRaw</code></summary>

The `encryptRaw` and `decryptRaw` functions seem to only be used within `crypto.ts`. If it's intended to only be used here, consider converting it to a nonexported toplevel function. For example,

```typescript
import { _ } from '@joplin/lib/locale';
import { Crypto, CryptoBuffer, Digest, CipherAlgorithm, EncryptionResult } from '@joplin/lib/services/e2ee/types';
import QuickCrypto from 'react-native-quick-crypto';
...more imports...

const encryptRaw = (data: CryptoBuffer, algorithm...more arguments...) => {
      // Implementation
};

const crypto: Crypto = {
     // Implementation
};

export default crypto;
```

This should prevent `encryptRaw` from being used where `encrypt` should be used instead.

**Ignore this suggestion if `encryptRaw` will be used outside `crypto.ts`**.

</details>

<details><summary>🔵 Optional/structure: Move parameters to an <code>options</code> object</summary> 

**🛈 Note 🛈**: If `encryptRaw` is made private to `crypto.ts` (and only used in one place), the change suggested below is probably unnecessary


With the current definition of `encryptRaw`, I could write incorrect code similar to the following:
```typescript
const data = ... ;
const algorithm = CipherAlgorithm.AES_256_GCM;
const key = ... ;
const iv = ... ;
crypto.encryptRaw(data, algorithm, iv, key, 16, null);
```

One way to make it more difficult to mix up `key` and `iv` when calling `encryptRaw` could be to use an options object for two or more parameters. 

For example,

```suggestion
	encryptRaw: ({ data, algorithm, key, iv, authTagLength, associatedData }: EncryptRawOptions) => {
```

`EncryptRawOptions` might then be defined:
```
interface EncryptRawOptions {
	data: CryptoBuffer;
	algorithm: CipherAlgorithm;
	key: CryptoBuffer;
	iv: CryptoBuffer | null;
	authTagLength: number;
	associatedData: CryptoBuffer | null;
}
```

The bad code above would then need to be written as follows:
```typescript
const data = ... ;
const algorithm = CipherAlgorithm.AES_256_GCM;
const key = ... ;
const iv = ... ;
crypto.encryptRaw({ data, algorithm, key: iv, iv: key, authTagLength: 16, associatedData: null });
```

</details>

Reply:
> Optional/structure: Don't export `encryptRaw`

I've removed the internal methods from the interface.

> Optional/structure: Move parameters to an `options` object

It's a good idea! Although I have made `encryptRaw()` private but I made a similar wrapper called `EncryptionParameters` to pass all the crypto parameters, so I can use the named arguments when calling `encrypt()`/`decyprt()`